app/services/authorization.rb

```diff
@@ -39,9 +39,11 @@ def projects(action, user)
     Authorization::ProjectQuery.query(user, action)
   end
 
-  # Returns all roles a user has in a certain project or globally
-  def roles(user, project = nil)
-    if project
+  # Returns all roles a user has in a certain project, for a specific entity or globally
+  def roles(user, project: nil, entity: nil)
```

Comment:
Personally, I am a fan of explicitly putting arguments in, instead of havin a catch-all argument that you can pass basically everything into. I understand that having the context is more aligned with the rest of the code. I'll refactor and try out how it feels.

Reply:
I have refactored it in d543cf3738. Let me know what you think